scripts/test-ci-all.sh

```diff
@@ -13,43 +13,42 @@ fi
 
 
 # Avoid re-building workspace in parallel in all test derivations
->&2 echo "### Making sure workspace is built..."
-nix build -L .#debug.workspaceBuild 2>&1 | ts -s
-
+# Note: Respect 'CARGO_PROFILE' that crane uses
+cargo build ${CARGO_PROFILE:+--profile ${CARGO_PROFILE}}
 
 function cli_test_reconnect() {
   set -eo pipefail # pipefail must be set manually again
   echo "### Starting reconnect test..."
-  nix build -L .#debug.cli-test.reconnect 2>&1 | ts -s
+  unshare -rn bash -c "ip link set lo up && exec unshare --user ./scripts/reconnect-test.sh" 2>&1 | ts -s
```

Comment:
so this creates a namespace with uid and guid mapped to 0 (-r), then creates a nework namespace (-n), sets the lo interface up and creates a new user namespace and runs the test script in it. cool.